src/Domain/DonationRepository.php

```diff
@@ -662,4 +666,48 @@ public function removeAllFundingWithdrawalsForDonation(Donation $donation): void
             }
         });
     }
+
+    private function setSalesforceIdIfNeeded(Donation $donation, string $salesforceId): void
+    {
+        if ($donation->getSalesforceId() !== null) {
```

Comment:
throw if the donation's existing sf ID is not null and different to the one we're trying to set?